packages/docprovider/src/yprovider.ts

```diff
@@ -110,14 +121,32 @@ export class WebSocketProvider implements IDocumentProvider {
     this._awareness.setLocalStateField('user', user.identity);
   }
 
+  private _onConnectionClosed = (event: any): void => {
+    if (event.code === 1003) {
+      console.error('Document provider closed:', event.reason);
+
+      showErrorMessage(
+        'Session expired',
+        'The document session expired. You need to reload this browser tab.',
+        [Dialog.okButton({ label: 'Reload' })]
+      )
+        .then(r => window.location.reload())
```

Comment:
I oversight this one in `jupyter_collaboration`. But we should be consistent and not reload if the user did not want that (for example that will happen if it clicks outside the dialog):

```suggestion
        .then(r => { 
          if (r.button.accept) {
            window.location.reload();
          }
        })
```